### PR TITLE
Fix flake8 warnings: Resolves issue #7

### PR DESCRIPTION
**Pull request by jriyyya**

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[flake8]
+
+# E402 module level import not at top of file
+# gi.require_version() is required before later imports
+
+# W503 line break before binary operator
+# Following the latest pep8 style Guide
+
+ignore = E402, W503
```
Comment (chimosky) [inline .flake8]: Could you remove the no new line at EOF?

File: lessons.py [hunk 2]
```diff
@@ -57,27 +59,32 @@ def __init__(self):
...
+
 _lessons_dir = os.path.join(get_bundle_path(), 'lessons')
 _lang = locale.getdefaultlocale()[0].split('_')[0]
 
 if not os.path.isdir(os.path.join(_lessons_dir, _lang)):
     logger.info('Cannot find lessons for language %s, thus use en' % _lang)
     _lang = 'en'
 
-for i, filename in enumerate(sorted(glob(os.path.join(_lessons_dir, _lang, '*')))):
+for i, filename in enumerate(sorted(glob(os.path.join(_lessons_dir,
```
Comment (chimosky) [inline lessons.py]: This is unreadable and sometimes some long lines are better left as long as they're readable as in this case there's quite the nested calls so not separating it right makes it unreadable.

I don't think this line should be split but if you wanted to do it, it's more readable having the `os.path` call on the new line so.

File: model.py [hunk 3]
```diff
@@ -127,12 +131,13 @@ def _make_thumbs(self):
...
         for key in self.scaled_sticks:
-            (angle,len) = self.scaled_sticks[key]
+            (angle, len) = self.scaled_sticks[key]
             newlen = int(len * .2)
-            self.scaled_sticks[key] = (angle,newlen)
+            self.scaled_sticks[key] = (angle, newlen)
 
         self._setjoints(self.scaled_joints, self.scaled_sticks,
-                (self.x, KEYFRAMEHEIGHT/2))
+                        (self.x, theme.KEYFRAMEHEIGHT / 2))
```
Comment (chimosky) [inline model.py:139]: Maybe use floor division here as you're dividing by an int but test to make sure the dimensions are as they should be so if `theme.KEYFRAMEHEIGHT` is  a float then no need to change it.

File: model.py [hunk 4]
```diff
@@ -142,74 +147,78 @@ def save(filename):
...
 
     for i, data in enumerate(inc):
         keys[i] = StoredFrame(data)
 
+
 def getparentsticks(stickname):
-    if stickname in ['RIGHT SHOULDER','LEFT SHOULDER','NECK','TORSO']:
+    if stickname in ['RIGHT SHOULDER', 'LEFT SHOULDER', 'NECK', 'TORSO']:
         return []
     if stickname in ['HEAD']:
```
Comment (chimosky) [inline model.py]: Not related to your change but could you change this check to `==` for consistency?

File: model.py [hunk 5]
```diff
@@ -223,8 +232,9 @@ def _get_base64_pixbuf_data(pixbuf):
     import base64
     return base64.b64encode(data[1])
 
+
 for i in range(5):
     key = StoredFrame()
-    keyframe_width = KEYFRAMEWIDTH/5
-    key.x = keyframe_width/2 + i*keyframe_width
+    keyframe_width = theme.KEYFRAMEWIDTH / 5
+    key.x = keyframe_width / 2 + i * keyframe_width
```
Comment (chimosky) [inline model.py:239]: Same comment as above about floor division.

File: montage.py [hunk 6]
```diff
@@ -166,29 +171,32 @@ def motion_notify_event(self, widget, event):
...
                     # add sum of parent angles to new angle
                     parents = model.getparentsticks(stickname)
                     panglesum = 0
                     for parentname in parents:
                         (pangle, plen) = self.key.sticks[parentname]
                         panglesum += pangle
                     (angle, len) = self.key.sticks[stickname]
-                    #print 'X:%s,Y:%s,PX:%s,PY:%s,ANGLE:%s,NEWANGLE:%s' % (x,y,px,py,angle,newangle)
+                    # print 'X:%s,Y:%s,PX:%s,PY:%s,ANGLE:%s,NEWANGLE:%s' %
+                    #       (x,y,px,py,angle,newangle)
```
Comment (chimosky) [inline montage.py]: You can just delete the comment.

File: montage.py [hunk 7]
```diff
@@ -209,7 +217,7 @@ def motion_notify_event(self, widget, event):
                 if _inarea(widget, x, y):
                     (px, py) = self.key.middle
                     if x - px == 0:
-                        #computeangle = 0
+                        # computeangle = 0
```
Comment (chimosky) [inline montage.py]: Also delete this too.

File: montage.py [hunk 8]
```diff
@@ -322,10 +334,10 @@ def playframe(self):
...
         self._draw_frame(self.playframenum, self.surface)
         # draw circle for middle
-        #green = cm.alloc_color('green')
-        #drawgc.set_foreground(green)
-        #x,y = middle
-        #self.pixmap.draw_arc(drawgc,True,x-5,y-5,10,10,0,360*64)
+        # green = cm.alloc_color('green')
+        # drawgc.set_foreground(green)
+        # x,y = middle
+        # self.pixmap.draw_arc(drawgc,True,x-5,y-5,10,10,0,360*64)
```
Comment (chimosky) [inline montage.py]: You can delete too, same for all other comments that aren't helpful.

File: montage.py [hunk 9]
```diff
@@ -415,154 +427,172 @@ def drawmainframe(self):
...
-        green = Gdk.Color.parse('green')[1]
-        pink = Gdk.Color.parse(PINK)[1]
-        bgcolor = Gdk.Color.parse(BACKGROUND)[1]
-        darkgreen = Gdk.Color.parse(BUTTON_BACKGROUND)[1]
+        # black = Gdk.Color.parse('black')[1]
+        # blue = Gdk.Color.parse('blue')[1]
+        # green = Gdk.Color.parse('green')[1]
+        # pink = Gdk.Color.parse(PINK)[1]
+        # bgcolor = Gdk.Color.parse(BACKGROUND)[1]
+        # darkgreen = Gdk.Color.parse(BUTTON_BACKGROUND)[1]
```
Comment (chimosky) [inline montage.py]: If these aren't used then delete them instead.

File: montage.py [hunk 10]
```diff
@@ -643,27 +677,30 @@ def __init__(self, activity):
...
                                | Gdk.EventMask.POINTER_MOTION_HINT_MASK)
-        self.mfdraw.set_size_request(DRAWWIDTH, DRAWHEIGHT)
+        self.mfdraw.set_size_request(theme.DRAWWIDTH, theme.DRAWHEIGHT)
 
         screen_box = Gtk.EventBox()
-        screen_box.modify_bg(Gtk.StateType.NORMAL, Gdk.color_parse(BACKGROUND))
-        screen_box.set_border_width(PAD / 2)
+        screen_box.modify_bg(Gtk.StateType.NORMAL,
+                             Gdk.color_parse(theme.BACKGROUND))
+        screen_box.set_border_width(theme.PAD / 2)
```
Comment (chimosky) [inline montage.py:660]: Same issue with floor division.

File: montage.py [hunk 11]
```diff
@@ -742,23 +786,27 @@ def __init__(self, activity):
...
         control_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
         control_box.pack_start(control_head, False, False, 0)
         control_box.pack_start(control_scroll, True, True, 0)
 
         control_bg = Gtk.EventBox()
-        control_bg.modify_bg(Gtk.StateType.NORMAL, Gdk.color_parse(BUTTON_BACKGROUND))
-        control_bg.set_border_width(PAD / 2)
+        control_bg.modify_bg(Gtk.StateType.NORMAL,
+                             Gdk.color_parse(theme.BUTTON_BACKGROUND))
+        control_bg.set_border_width(theme.PAD / 2)
```
Comment (chimosky) [inline montage.py:778]: Floor division...

File: montage.py [hunk 12]
```diff
@@ -781,26 +829,27 @@ def __init__(self, activity):
         desktop.pack_start(kfdraw_box, False, False, 0)
 
         greenbox = Gtk.EventBox()
-        greenbox.modify_bg(Gtk.StateType.NORMAL, Gdk.color_parse(BACKGROUND))
-        greenbox.set_border_width(PAD / 2)
+        greenbox.modify_bg(Gtk.StateType.NORMAL,
+                           Gdk.color_parse(theme.BACKGROUND))
+        greenbox.set_border_width(theme.PAD / 2)
```
Comment (chimosky) [inline montage.py:809]: Here too.

File: screenflip.py [hunk 13]
```diff
@@ -26,7 +27,7 @@ def reset(self):
         self.parts = theme.PARTS.copy()
         self.sticks = theme.STICKS.copy()
         self.joints = self._initjoints()
-        self.middle = (theme.DRAWWIDTH/2, theme.DRAWHEIGHT/3)
+        self.middle = (theme.DRAWWIDTH / 2, theme.DRAWHEIGHT / 3)
```
Comment (chimosky) [inline screenflip.py:30]: floor division check too.

File: theme.py [hunk 14]
```diff
@@ -18,70 +18,72 @@
 
 from sugar3.graphics import style
 
-GRAY = "#B7B7B7" # gray
-PINK = "#FF0099" # pink
-YELLOW = "#FFFF00" # yellow
+GRAY = "#B7B7B7"  # gray
+PINK = "#FF0099"  # pink
+YELLOW = "#FFFF00"  # yellow
```
Comment (chimosky) [inline theme.py]: You can remove the comments here as the variable names do their job.

File: theme.py [hunk 15]
```diff
@@ -18,70 +18,72 @@
...
-    (Gtk.StateType.PRELIGHT,"#016D01"),
-    (Gtk.StateType.SELECTED,"#CCFF99"),
-    (Gtk.StateType.INSENSITIVE,"#027F01"),
-    )
+    (Gtk.StateType.NORMAL, "#027F01"),
+    (Gtk.StateType.ACTIVE, "#CCFF99"),
+    (Gtk.StateType.PRELIGHT, "#016D01"),
+    (Gtk.StateType.SELECTED, "#CCFF99"),
+    (Gtk.StateType.INSENSITIVE, "#027F01"),
+)
```
Comment (chimosky) [inline theme.py]: This isn't related to your change but could you use variable names in place of the color codes for the ones that have been defined in the lines right above this.

File: theme.py [hunk 16]
```diff
@@ -18,70 +18,72 @@
...
-            'LEFT HIP'          : _('Left Hip'),
-            'UPPER LEFT LEG'    : _('Upper Left Leg'),
-            'LOWER LEFT LEG'    : _('Lower Left Leg'),
-            'LEFT FOOT'         : _('Left Foot') }
+    (Gtk.StateType.NORMAL, "#027F01"),
+    (Gtk.StateType.ACTIVE, "#014D01"),
+    (Gtk.StateType.PRELIGHT, "#016D01"),
+    (Gtk.StateType.SELECTED, "#027F01"),
+    (Gtk.StateType.INSENSITIVE, "#027F01"),
+)
```
Comment (chimosky) [inline theme.py]: Same comment as above.

File: theme.py [hunk 17]
```diff
@@ -92,101 +94,144 @@
...
+          'LOWER LEFT ARM': 'lefthand',
+          'TORSO': 'groin',
+          'RIGHT HIP': 'righthip',
+          'UPPER RIGHT LEG': 'rightknee',
+          'LOWER RIGHT LEG': 'rightheel',
+          'RIGHT FOOT': 'righttoe',
+          'LEFT HIP': 'lefthip',
+          'UPPER LEFT LEG': 'leftknee',
+          'LOWER LEFT LEG': 'leftheel',
+          'LEFT FOOT': 'lefttoe'}
```
Comment (chimosky) [inline theme.py]: Would be great to be consistent like the dicts before this one.

File: toolkit/activity.py [hunk 18]
```diff
@@ -284,8 +281,9 @@ def _shared_cb(self, activity):
         self._sharing_setup()
 
         logging.debug('This is my activity: making a tube...')
-        id = self._tubes_chan[TelepathyGLib.IFACE_CHANNEL_TYPE_TUBES].OfferDBusTube(
-            self.service, {})
+        # id = self._tubes_chan[
+        #     TelepathyGLib.IFACE_CHANNEL_TYPE_TUBES].OfferDBusTube(
+        #     self.service, {})
```
Comment (chimosky) [inline toolkit/activity.py]: You can delete if it isn't used.

**Review by chimosky — COMMENTED** [review]
Reviewed, not tested.

For the floor division comments, you can leave the changes as this isn't a functional change PR.

File: lessons.py (hunk 2 above)
Comment (jriyyya) [reply lessons.py]: Done.

File: model.py (hunk 4 above)
Comment (jriyyya) [reply model.py]: Done.

File: montage.py (hunk 6 above)
Comment (jriyyya) [reply montage.py]: Done.

File: theme.py (hunk 15 above)
Comment (jriyyya) [reply theme.py]: Done.

**Comment by jriyyya** [comment]
Hi @chimosky , Made changes as per the review, will open a new PR for floor divison as you said. I was confused about what you wanted for the consistency in the code in the file `Theme.py` about the dicts, but later understood, so the last commit is related to it. 

I was going to squash these commits in a single commit but after the commits are pushed, they can't be squashed 

Sorry for the mess.

File: .flake8 (hunk 1 above)
Comment (jriyyya) [reply .flake8]: I am not able to understand what is to be done. 

Is the no new line at EOF needs to be added in ignore?

File: model.py (hunk 4 above)
Comment (jriyyya) [reply model.py]: Does this change  to be done just in `stickname in ['HEAD']` or with `if stickname in ['RIGHT SHOULDER', 'LEFT SHOULDER', 'NECK', 'TORSO']` too making 4 different if conditions for all

File: .flake8 (hunk 1 above)
Comment (chimosky) [reply .flake8]: Removing the new line at EOF makes it a bit difficult to view when using tools like cat as there's no new line at the end so the file gets printed to the output and the prompt shows right where the file ends, try this and see the difference;

```
# check the output of .flake8 at your latest change and you'll see what I mean
$ cat .flake8 
# This fixes it though
$ echo "" >> .flake8
$ cat .flake8
```

You'll notice the difference.

Also checkout out the first part of our [contributing](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#contributing) doc, changes are to be made in a separate branch on your fork before opening a PR to the master/main branch.

File: model.py (hunk 4 above)
Comment (chimosky) [reply model.py]: Just `[HEAD]` as it's just one item and to be consistent with other lines where it's just one item.

File: lessons.py [hunk 19]
```diff
@@ -85,6 +85,6 @@ def __init__(self):
     logger.info('Cannot find lessons for language %s, thus use en' % _lang)
     _lang = 'en'
 
-for i, filename in enumerate(sorted(glob(os.path.join(_lessons_dir,
-                                                      _lang, '*')))):
+for i, filename in enumerate(sorted(glob(
+    os.path.join(_lessons_dir, _lang, '*')))):
```
Comment (chimosky) [inline lessons.py]: Isn't readable as it's on the same line as the code block in the if statement and it's not supposed to be as they're two different blocks.

**Comment by chimosky** [comment]
Reviewed e3737b4, could you not force push as this is a flake8 PR and every commit shows the progress of the PR?

File: .flake8 (hunk 1 above)
Comment (quozl) [reply .flake8]: I agree with @chimosky, please fix the "No newline at end of file" warning.  If you don't know how to press enter in the right place, please tell us under what cursed development environment you are enslaved, and we can giggle quietly to ourselves or give better advice.  :grinning:

Comment (jriyyya) [reply .flake8]: @quozl I am sorry about that, Fixing it..

**Comment by jriyyya** [comment]
@chimosky I was doubtful about the earlier  commits and deemed them to be incorrect and thus had to force push back to the first commit, so that I could make amends as needed to the commit messages, had to make a few new commits as well.

File: .flake8 (hunk 1 above)
Comment (jriyyya) [reply .flake8]: > Also checkout out the first part of our [contributing](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#contributing) doc, changes are to be made in a separate branch on your fork before opening a PR to the master/main branch.

for this PR, Continuing in the same branch, would be making new branch for future PRs

**Comment by quozl** [comment]
Reviewed to 8f06986.

File: .flake8 (hunk 1 above)
Comment (chimosky) [reply .flake8]: > @quozl I am sorry about that, Fixing it..

There's nothing to be sorry for, you didn't do anything wrong and Quozl was only making a joke.

**Comment by chimosky** [comment]
> @chimosky I was doubtful about the earlier commits and deemed them to be incorrect and thus had to force push back to the first commit, so that I could make amends as needed to the commit messages, had to make a few new commits as well.

Yes I understand and I'm saying that for changes sometimes there's no need to reset and force push as in this case, the commits just show the progress you made while working on this.

**Comment by quozl** [comment]
Agreed.  For changing commit messages, you certainly need to force push, as the message forms input to the git hash.  GitHub provides a link "force-pushed" to the differences, but it is best to restrict yourself to commit message changes when doing a force push while under review by others.  Otherwise review is made more difficult.

**Comment by jriyyya** [comment]
Okayy Got it, I will keep this in mind

**Pull request merged**